test/BUILD

```diff
@@ -0,0 +1,44 @@
+load("@rules_cc//cc:defs.bzl", "cc_test")
+load("//:variables.bzl", "COPTS", "LINKOPTS")
+
+cc_test(
+    name = "null_vm_test",
+    srcs = ["null_vm_test.cc"],
+    copts = COPTS,
+    linkopts = LINKOPTS,
+    deps = [
+        "//:lib",
+        "@com_google_googletest//:gtest",
+        "@com_google_googletest//:gtest_main",
+    ],
+)
+
+cc_test(
+    name = "wasmtime_vm_test",
+    srcs = ["wasmtime_vm_test.cc"],
+    copts = COPTS,
+    linkopts = LINKOPTS,
+    deps = [
+        "//:lib",
+        "//test:watdata",
+        "@com_google_googletest//:gtest",
+        "@com_google_googletest//:gtest_main",
+    ],
+)
+
+cc_test(
+    name = "context_test",
+    srcs = ["context_test.cc"],
+    copts = COPTS,
+    linkopts = LINKOPTS,
+    deps = [
+        "//:include",
+        "@com_google_googletest//:gtest",
+        "@com_google_googletest//:gtest_main",
+    ],
+)
+
+cc_library(
+    name = "watdata",
+    hdrs = ["watdata.h"],
+)
```

Comment:
Note that the support for text format is not universal across runtimes (notably, V8 cannot execute it), so using it in test data is generally discouraged, and we try to write tests data using one of the existing SDKs.

Reply:
these wat strings are converted to native wasm binary in the setup time of each test case like this: https://github.com/proxy-wasm/proxy-wasm-cpp-host/blob/2a2b2ef41c185f73d739edc1fda684ace8bb1748/test/wasmtime_vm_test.cc#L64-L74

I think this is universal enough for testing any runtime as long as we use `wasmtime`'s api.

Reply:
But we won't be able to use it together with V8 or WAVM runtimes due to linking issues.

Reply:
That's true. Thanks, I'll just use Rust

Reply:
You could also use your Proxy-Wasm Go SDK, I don't think we have any tests against it.

Reply:
resolved in https://github.com/proxy-wasm/proxy-wasm-cpp-host/pull/73/commits/abae5f08df6731536a0f7d820bba8c1b3ed0c1d0

As for Go SDK, it's going to be tricky since it's not based on official compiler. so I'll put that in my TODO list and come back to latter